Hibernate initalization issue with Multiple Datasources and Compose Dev Services

### Describe the bug

I am trying to create a simple example with Hibernate ORM Panache, Multiple Datasources and Compose Dev Services.
I can spin up the containers and see exactly the default datasource in the console with the example provided here: https://github.com/myfear/ejq_substack_articles/tree/main/xa-transaction-tutorial

As soon as I uncomment the following two lines in the application.properties:

# --- Hibernate ORM configuration ---
#=com.example.customer
#quarkus.hibernate-orm.accounts.packages=com.example.account

- The Compose Dev Services do not start the containers anymore. (podman ps lists zero containers)
- And I get below exception:

```posh
2025-07-30 08:41:58,352 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor#configurationDescriptorBuilding threw an exception: io.quarkus.runtime.configuration.ConfigurationException: Datasource must be defined for persistence unit 'accounts'. Refer to https://quarkus.io/guides/datasource for guidance.
	at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.collectDialectConfig(HibernateOrmProcessor.java:997)
	at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.producePersistenceUnitDescriptorFromConfig(HibernateOrmProcessor.java:949)
	at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.handleHibernateORMWithNoPersistenceXml(HibernateOrmProcessor.java:889)
	at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.configurationDescriptorBuilding(HibernateOrmProcessor.java:352)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:372)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:289)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:61)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:89)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:432)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:143)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:98)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:107)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:70)
```


### Expected behavior

Both containers start and Hibernate initializes accordingly.

### Actual behavior

No Dev Service container is started and I get the mentioned exception.

### How to Reproduce?

https://github.com/myfear/ejq_substack_articles/tree/main/xa-transaction-tutorial

1) Checkout and start Quarkus => Containers start as expected. => Stop Quarkus
2) Uncomment Lines 17+18 here https://github.com/myfear/ejq_substack_articles/blob/main/xa-transaction-tutorial/src/main/resources/application.properties
3) Start Quarkus => Exception

### Output of `uname -a` or `ver`

Darwin Kernel Version 24.5.0: Tue Apr 22 19:53:27 PDT 2025; root:xnu-11417.121.6~2/RELEASE_ARM64_T6041 arm64

### Output of `java -version`

openjdk 21.0.7 2025-04-15 LTS OpenJDK Runtime Environment Temurin-21.0.7+6 (build 21.0.7+6-LTS) OpenJDK 64-Bit Server VM Temurin-21.0.7+6 (build 21.0.7+6-LTS, mixed mode, sharing)

### Quarkus version or git rev

 <quarkus.platform.version>3.24.5</quarkus.platform.version>

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937)

### Additional information

_No response_

I assume that setting:

```
quarkus.hibernate-orm.accounts.datasource=accounts
```

should work (see the [docs](https://quarkus.io/guides/hibernate-orm#hibernate-dialect-varying-database) for more details). Can you try?

@yrodiere would it not make sense to change the error message from:

```
Caused by: io.quarkus.runtime.configuration.ConfigurationException: Datasource must be defined for persistence unit 'accounts'. Refer to https://quarkus.io/guides/datasource for guidance.               
```

to something more actionable like:

```
Caused by: io.quarkus.runtime.configuration.ConfigurationException: Datasource must be defined for persistence unit 'accounts'. Setting the datasource for the persistence unit can be done via the `quarkus.hibernate-orm.accounts.datasource` property. Refer to https://quarkus.io/guides/hibernate-orm#hibernate-dialect-varying-database and https://quarkus.io/guides/datasource for all the details.               
```

--- a/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
+++ b/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
@@ -989,10 +989,11 @@ public final class HibernateOrmProcessor {
         Optional<String> dbKind = jdbcDataSource.map(JdbcDataSourceBuildItem::getDbKind);
         Optional<String> explicitDbMinVersion = jdbcDataSource.flatMap(JdbcDataSourceBuildItem::getDbVersion);
         if (multiTenancyStrategy != MultiTenancyStrategy.DATABASE && jdbcDataSource.isEmpty()) {
+            String dsConfigProperty = HibernateOrmRuntimeConfig.puPropertyKey(persistenceUnitName, "datasource");
             throw new ConfigurationException(String.format(Locale.ROOT,
-                    "Datasource must be defined for persistence unit '%s'."
+                    "Datasource must be defined for persistence unit '%s'. Setting the datasource for the persistence unit can be done via the '%s' property. "
                             + " Refer to https://quarkus.io/guides/datasource for guidance.",
-                    persistenceUnitName),
+                    persistenceUnitName, dsConfigProperty),
                     new HashSet<>(Arrays.asList("quarkus.datasource.db-kind", "quarkus.datasource.username",
                             "quarkus.datasource.password", "quarkus.datasource.jdbc.url")));
         }

--- a/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/config/datasource/EntitiesInNamedPUWithoutDatasourceTest.java
+++ b/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/config/datasource/EntitiesInNamedPUWithoutDatasourceTest.java
@@ -23,7 +23,8 @@ public class EntitiesInNamedPUWithoutDatasourceTest {
             .overrideConfigKey("quarkus.hibernate-orm.pu-1.schema-management.strategy", "drop-and-create")
             .assertException(t -> assertThat(t)
                     .isInstanceOf(ConfigurationException.class)
-                    .hasMessageContainingAll("Datasource must be defined for persistence unit 'pu-1'."));;
+                    .hasMessageContainingAll("Datasource must be defined for persistence unit 'pu-1'.")
+                    .hasMessageContainingAll("'quarkus.hibernate-orm.\"pu-1\".datasource'"));
 
     @Test
     public void testInvalidConfiguration() {

